draft-ietf-tls-esni.md

```diff
@@ -791,6 +793,13 @@ where HKDF-Expand-Label and HKDF-Extract are as defined in {{RFC8446}}. The
 value of ServerHello.random[0:24] is generated as usual by invoking a secure
 random number generator (see {{RFC8446}}, Section 4.1.2).
 
+\[\[TODO: If the ClientHello offers TLS 1.2 or below, it's possible the backend
+server will negotiate it and leak the server name via an unencrypted server
+certificate. This shouldn't happen because clients are forbidden from offering
+earlier versions, but perhaps either the backend server should check (easy in
+Shared Mode, hard in Split Mode), or maybe the client-facing server should
```

Comment:
Why is this hard in Split Mode? We could mandate that if the backend server observes the empty ECH extension (sent in the ClientHelloInner in order to signal the backend server to confirm acceptance), it aborts unless it can negotiate 1.3 or higher.